### PR TITLE
Restore legacy onion (damn you lnd!)

### PR DESCRIPTION
**Pull request by rustyrussell**
Let's remove it again in... um...

Shit, it's going to be there forever, isn't it?

:sob: 

Fixes: #7347

### CODE REVIEW

**Review by vincenzopalazzo — APPROVED** [review]
I was going to preferer some git revert commit in this way we can revert the revert operation at some point :P 

>Shit, it's going to be there forever, isn't it?

If we remove it, what will we pass down to future generations? 😹


ACK https://github.com/ElementsProject/lightning/pull/7352/commits/e0d87353215dc58451b2700b94f90df6755d4216

File: lightningd/pay.c [hunk 1]
```diff
@@ -1177,6 +1178,14 @@ send_payment(struct lightningd *ld,
 		ret = pubkey_from_node_id(&pubkey, &ids[i]);
 		assert(ret);
 
+		if (dev_legacy_hop && i == n_hops - 2) {
+			sphinx_add_v0_hop(path, &pubkey,
+					  &route[i + 1].scid,
+					  route[i + 1].amount,
+					  base_expiry + route[i + 1].delay);
+			continue;
```
Comment (cdecker) [inline lightningd/pay.c:1186]: Just a stylistic thing, are you avoiding `if {} else {}` for legacy and non-legacy to minimize the diff when we remove legacy again?

**Review by cdecker — APPROVED** [review]
Looking good, too bad they haven't removed the legacy version yet.

ACK https://github.com/ElementsProject/lightning/pull/7352/commits/e0d87353215dc58451b2700b94f90df6755d4216

**Review by endothermicdev — APPROVED** [review]
ACK e0d87353215dc58451b2700b94f90df6755d4216

**Pull request merged**